main.go

```diff
@@ -338,15 +329,24 @@ func main() {
 
 		os.Exit(0)
 	}
-	ctrl.Run(ctx, stopChan)
+
+	if cfg.UpdateEvents {
+		// Add RunOnce as the handler function that will be called when ingress/service sources have changed.
+		// Note that k8s Informers will perform an initial list operation, which results in the handler
+		// function initially being called for every Service/Ingress that exists
+		ctrl.Source.AddEventHandler(ctx, func() { ctrl.ScheduleRunOnce(time.Now()) })
```

Comment:
Now that I look at this? This looks like a case of an implementation feeding itself

Reply:
I'm not sure what you mean. This is to trigger ScheduleRunOnce event when something changes in Source

Reply:
What I mean is that we are passing a field of `ctrl` as a parameter to a method of `ctrl`. Hence the "feeding itself" idiom. But this was already existing, so I wouldn't say this is very important

Reply:
I can refactor it in separate PR :)